src/main/java/uk/gov/pay/directdebit/mandate/dao/MandateDao.java

```diff
@@ -121,6 +121,12 @@ int updateStateByPaymentProviderMandateId(@Bind("provider") PaymentProvider paym
                                               @Bind("providerMandateId") PaymentProviderMandateId paymentProviderMandateId,
                                               @Bind("state") MandateState mandateState);
 
+    @SqlUpdate("UPDATE mandates m SET state = :state FROM gateway_accounts g WHERE m.payment_provider_id = :providerMandateId " +
+            "AND g.id = m.gateway_account_id AND g.organisation IS NULL AND g.payment_provider = :provider")
+    int updateStateByPaymentProviderMandateId(@Bind("provider") PaymentProvider paymentProvider,
```

Comment:
Perhaps rename the existing `updateStateByPaymentProviderMandateId` to something like `updateStateByPaymentProviderMandateIdAndServiceId` to help make the distinction between these two methods?

Reply:
Renamed to `updateStateByPaymentProviderMandateIdAndOrganisationId` (since it’s always going to be a GoCardless organisation ID)